utils/get_test_info.py

```diff
@@ -0,0 +1,118 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+import os
+import sys
+import unittest
+
+
+# This is required to make the module import works (when the python process is running from the root of the repo)
+sys.path.append(".")
+
+
+r"""
+The argument `test_file` in this file refers to a model test file. This should be a string of the from
+`tests/models/*/test_modeling_*.py`.
+"""
+
+
+def get_module_path(test_file):
+    """Return the module path of a model test file."""
+    components = test_file.split(os.path.sep)
+    if components[0:2] != ["tests", "models"]:
+        raise ValueError(
+            "`test_file` should start with `tests/models/` (with `/` being the OS specific path separator). Got "
+            f"{test_file} instead."
+        )
+    test_fn = components[-1]
+    if not test_fn.endswith("py"):
+        raise ValueError(f"`test_file` should be a python file. Got {test_fn} instead.")
+    if not test_fn.startswith("test_modeling_"):
+        raise ValueError(
+            f"`test_file` should point to a file name of the form `test_modeling_*.py`. Got {test_fn} instead."
+        )
+
+    components = components[:-1] + [test_fn.replace(".py", "")]
+    test_module_path = ".".join(components)
+
+    return test_module_path
+
+
+def get_test_module(test_file):
+    """Get the module of a model test file."""
+
+    test_module_path = get_module_path(test_file)
+    test_module = importlib.import_module(test_module_path)
+
+    return test_module
+
+
+def get_tester_classes(test_file):
+    """Get all classes in a model test file whose names ends with `ModelTester`."""
+
+    tester_classes = []
+    test_module = get_test_module(test_file)
+    for attr in dir(test_module):
+        if attr.endswith("ModelTester"):
+            tester_classes.append(getattr(test_module, attr))
+
+    return tester_classes
+
+
+def get_test_classes(test_file):
+    """Get all test classes in a model test file which have attribute `all_model_classes`.
+
+    These are the (model) test classes containing the (non-slow) tests to run. They should be a subclass of one of the
+    classes `ModelTesterMixin`, `TFModelTesterMixin` or `FlaxModelTesterMixin`, as well as a subclass of
+    `unittest.TestCase`.
+
+    Their `all_model_classes` attributes should be non-empty tuples, except a few rare cases.
+    """
+    test_classes = []
+    test_module = get_test_module(test_file)
+    for attr in dir(test_module):
+        attr_value = getattr(test_module, attr)
+        if type(attr_value) == type:
+            if issubclass(attr_value, unittest.TestCase) or attr_value.__name__ in ["RagTestMixin", "TFRagTestMixin"]:
```

Comment:
These are edge cases, where they have `all_model_classes` but not `unittest.TestCase`.

Reply:
Isn't it more efficient to test if `attr_value` has an attribute named `"all_model_classes"` (in case we get other models like this in the future)?

Reply:
It depends on the situations. In the demo usage I made for check repo , we focus on this attribute. In other case, what we want is just to get model tester class. But you might be right, even in this case, it will only be useful when we have this all_model_classes attribute. I am happy go for the suggestion 🙂